app/models/supermarket.rb

```diff
@@ -1,3 +1,7 @@
 class Supermarket < ApplicationRecord
   has_many :items
+
+  def list_customers
+    Customer.joins(items: :customeritems).where('items.supermarket_id' => id).distinct
+  end
```

Comment:
This is a great joins, and works perfectly. An easier way to have done the same thing without having to go through a joins would be to add an association.

```ruby
has_many :customers, through: :items
```

The reason we can do this is because `items` already has an association with `customers` through `customer_items`.

You would then be able to make use of that association in your `list_customers` method like so:

```ruby
def list_customers
  customers.distinct
end
```

I would also recommend changing the method name to `list_unique_customers` so it's obvious what the method is doing.